src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -228,14 +152,21 @@ JNI_METHOD(jlong, newDeviceController)
 
     ChipLogProgress(Controller, "newDeviceController() called");
 
-    DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().InitializeWithObject(keyValueStoreManager);
-    using ::chip::Mdns::InitializeWithObjects;
-    InitializeWithObjects(serviceResolver, chipMdnsCallback);
+    // //move sSystemLayer and sInetLayer into platform/android to share with app server
+    err = DeviceLayer::PlatformMgr().InitChipStack();
+    SuccessOrExit(err);
 
     wrapper = AndroidDeviceControllerWrapper::AllocateNew(sJVM, self, JniReferences::GetInstance().GetStackLock(), kLocalDeviceId,
-                                                          &sSystemLayer, &sInetLayer, &err);
+                                                          &DeviceLayer::SystemLayer(), &DeviceLayer::InetLayer, &err);
     SuccessOrExit(err);
 
+    // Create and start the IO thread. Must be called after Controller()->Init
+    if (sIOThread == PTHREAD_NULL)
```

Comment:
The device layer IO thread are in src/include/platform/internal/GenericPlatformManagerImpl_POSIX.cpp, there is no AttachCurrentThreadAsDaemon in it. Or do you think we should rewite _StartEventLoopTask in src/platform/android/PlatformManagerImpl.cpp?

Reply:
No, because we are using RunEventLoop ( which means create the IO Thread out of PlatformManagerImpl), instead of StartEventLoopTask ( Which means create IO Thread inside PlatformManagerImpl). It supports both way.

Reply:
I see. Thanks for the explanation!